File: problems/771/solution.ts
// Accepted: 46ms (97.55%), 51.50MB (14.48%)

function numJewelsInStones(jewels: string, stones: string): number {
    const counts = Array("z".charCodeAt(0)+1).fill(0);
    for (const stone of stones) {
        ++counts[stone.charCodeAt(0)];
    }

    const numJewels = Array.from(jewels)
        .map(jewel => counts[jewel.charCodeAt(0)])
        .reduce((a, b) => a + b);
    return numJewels;
};

File: problems/771/solution.py
# Accepted: 34ms (78.47%), 16.48MB (65.36%)

class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        counts = [0] * (ord('z')+1)
        for stone in stones:
            counts[ord(stone)] += 1

        numJewels = sum(counts[ord(jewel)] for jewel in jewels)
        return numJewels
